hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/HoodieTimelineArchiver.java

```diff
@@ -509,7 +509,15 @@ private Stream<HoodieInstant> getCommitInstantsToArchive() throws IOException {
   }
 
   private Stream<HoodieInstant> getInstantsToArchive() throws IOException {
-    Stream<HoodieInstant> instants = Stream.concat(getCleanInstantsToArchive(), getCommitInstantsToArchive());
+    List<HoodieInstant> commitInstantsToArchive = getCommitInstantsToArchive().collect(Collectors.toList());
+    Stream<HoodieInstant> instants = Stream.concat(getCleanInstantsToArchive(), commitInstantsToArchive.stream());
+    HoodieInstant hoodieOldestInstantToArchive = commitInstantsToArchive.stream().max(Comparator.comparing(maxInstant -> maxInstant.getTimestamp())).orElse(null);
+    /**
+     * if hoodieOldestInstantToArchive is null that means nothing is getting archived, so no need to update metadata
+     */
+    if (hoodieOldestInstantToArchive != null) {
+      table.getIndex().updateMetadata(table, Option.of(hoodieOldestInstantToArchive));
```

Comment:
I don't think invoking `updateMetadata` in `getInstantsToArchive` make sense.

Reply:
Okay, so from where this function should get called ??

Reply:
**getInstantsToArchive** is where archival process getting list of commit to archives. What issues you see for making the update **metadata call after getInstantsToArchive** ?

Reply:
@rohan-uptycs, `getInstantsToArchive` is also used to get instants to archive, therefore the update behavior couldn't invoke in this method for design.

Reply:
@SteNicholas , understood your concern. Should we run update metadata in postcommit of clustering operation?   https://github.com/apache/hudi/blob/b690346a700121124e2da8eb0013674ba7a0d719/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java#L522. 

Will that makes sense??
As new metadata file will get created only with clustering operation in consistent hash engine.

Reply:
@rohan-uptycs, make sense to me from interface point of view.

Reply:
@SteNicholas , using postCommit has some flaws and concerns. if postcommit operation fails due to some reason, then metadata sync will not happen , and metadata state will remain in in-consistent state. 
But in case of archival it will get updated in next archival trigger.

Reply:
@SteNicholas , Yeah it can be invoked but i see few problems with it 
What if underlying file system is down and **updateMetadata** fails to sync metadata, then there is no  mechanism to bring it in sync with latest committed metadata, and archival will remove replace commit eventually and it will end up in an inconsistent state.
On the other hand in **archival process , it will be eventually in sync with committed metadata**  before replace commit getting archived.
I think consistent hashing metadata has strong dependency on archival process, As it is dependent on active timeline replaced commit to load metadata.

Reply:
If we rename interface method as **updateArchivalDependentIndexMetadata** and call it in **archiveIfRequired** method will that be okay ?

Reply:
@rohan-uptycs, make senses to me.